lib/main_common.pm

```diff
@@ -138,8 +138,11 @@ sub init_main {
 
 sub loadtest {
     my ($test, %args) = @_;
-    croak "extensions are not allowed here '$test'" if $test =~ /\.pm$/;
-    autotest::loadtest("tests/$test.pm", %args);
+    if ($test =~ /\.p[my]$/) {
```

Comment:
This looks good!
You could also combine the if/else into the load test call:

```
autotest::loadtest('tests/' . ($test =~ /\.p[my]$/ ? $test : "$test.pm"), %args);

Reply:
Thanks!